### ISSUE TITLE
In Cisco DNA Center Reserved Sandbox, Cannot deploy template as getting error status: "FAILURE" 

### ISSUE THREAD

**Issue by Pratik993** [CiscoDevNet/DNAC-TemplateProgrammer#2]
Dear Team,

As a beginner, i am trying to deploy template on Cisco DNA Center reserved Sandbox,

VLAN on particular interface.
Ended getting status as FAILURE
Error message on under:

python template.py --template HWI/hs-wismar --device 10.10.20.84 --params "{ \"vlan\": \"20\", \"interface\": \"FastEthernet1/0/1\"}"
Looking for: HWI/hs-wismar
https://10.10.20.85:443/dna/intent/api/v1/template-programmer/template
TemplateId: a6bfe9b7-1303-4aa3-91ce-cb3e2c3d1595 Version: 1

https://10.10.20.85:443/dna/intent/api/v1/template-programmer/template/a6bfe9b7-1303-4aa3-91ce-cb3e2c3d1595
Showing Template Body:
interface $interface
  switchport access vlan $vlan

Required Parameters for template body: {"vlan":"","interface":""}

Bindings []

Executing template on:10.10.20.84, with Params:{ "vlan": "20", "interface": "FastEthernet1/0/1"}
payload {"templateId": "a6bfe9b7-1303-4aa3-91ce-cb3e2c3d1595", "forcePushTemplate": false, "targetInfo": [{"id": "10.10.20.84", "type": "MANAGED_DEVICE_IP", "params": {"vlan": "20", "interface": "FastEthernet1/0/1"}}]}
https://10.10.20.85:443/dna/intent/api/v1/template-programmer/template/deploy
Response {'deploymentId': 'Deployment of  Template: a6bfe9b7-1303-4aa3-91ce-cb3e2c3d1595.ApplicableTargets: [10.10.20.84]Template Deployemnt Id: d87ff5f0-226e-49ff-b219-7cf261635671', 'startTime': '', 'endTime': '', 'duration': '0 seconds'}
waiting for deploymentId d87ff5f0-226e-49ff-b219-7cf261635671
https://10.10.20.85:443/api/v1/template-programmer/template/deploy/status/d87ff5f0-226e-49ff-b219-7cf261635671
{'deploymentId': 'd87ff5f0-226e-49ff-b219-7cf261635671', 'templateName': 'hs-wismar', 'templateVersion': '1', 'projectName': 'HWI', 'status': 'INIT', 'startTime': '', 'endTime': '', 'duration': '0 seconds', 'devices': [{'deviceId': '12f826f5-5eb9-48d2-a4df-ec3756b4ef48', 'name': '', 'status': 'IN_PROGRESS', 'detailedStatusMessage': 'Provisioning  failed for the template hs-wismar.Message: Connection to device 10.10.20.84 timed out using protocol ssh2; Detail: FAILED : Error in running XDE Procedure, Error Message::Failed to establish ssh2 connection to device- Cause : Connection refused or timed-out.', 'startTime': '07:15:44 08/09/2019', 'endTime': '', 'ipAddress': '12f826f5-5eb9-48d2-a4df-ec3756b4ef48', 'duration': '0 minutes 2 seconds'}]}
Task=d87ff5f0-226e-49ff-b219-7cf261635671 has not completed yet. Sleeping 2 seconds...
https://10.10.20.85:443/api/v1/template-programmer/template/deploy/status/d87ff5f0-226e-49ff-b219-7cf261635671
{'deploymentId': 'd87ff5f0-226e-49ff-b219-7cf261635671', 'templateName': 'hs-wismar', 'templateVersion': '1', 'projectName': 'HWI', 'status': 'FAILURE', 'startTime': '', 'endTime': '07:15:48 08/09/2019', 'duration': '0 seconds', 'devices': [{'deviceId': '12f826f5-5eb9-48d2-a4df-ec3756b4ef48', 'name': '', 'status': 'FAILURE', 'detailedStatusMessage': 'Provisioning  failed for the template hs-wismar.Message: Connection to device 10.10.20.84 timed out using protocol ssh2; Detail: FAILED : Error in running XDE Procedure, Error Message::Failed to establish ssh2 connection to device- Cause : Connection refused or timed-out.', 'startTime': '07:15:44 08/09/2019', 'endTime': '', 'ipAddress': '12f826f5-5eb9-48d2-a4df-ec3756b4ef48', 'duration': '0 minutes 7 seconds'}]}

Response:
{
  "deploymentId": "d87ff5f0-226e-49ff-b219-7cf261635671",
  "templateName": "hs-wismar",
  "templateVersion": "1",
  "projectName": "HWI",
  "status": "FAILURE",
  "startTime": "",
  "endTime": "07:15:48 08/09/2019",
  "duration": "0 seconds",
  "devices": [
    {
      "deviceId": "12f826f5-5eb9-48d2-a4df-ec3756b4ef48",
      "name": "",
      "status": "FAILURE",
      "detailedStatusMessage": "Provisioning  failed for the template hs-wismar.Message: Connection to device 10.10.20.84 timed out using protocol ssh2; Detail: FAILED : Error in running XDE Procedure, Error Message::Failed to establish ssh2 connection to device- Cause : Connection refused or timed-out.",
      "startTime": "07:15:44 08/09/2019",
      "endTime": "",
      "ipAddress": "12f826f5-5eb9-48d2-a4df-ec3756b4ef48",
      "duration": "0 minutes 7 seconds"
    }
  ]
}

Anyone have solution for this?

i will be very thankful,.

**Comment by Pratik993** [comment]
Dear @aradford123 Sir,

Can you help me to solve above mentioned error?

**Comment by aradford123** [comment]
The issue seems to be an error with DNAC communication to the device.  It is not related to the template API.

detailedStatusMessage": "Provisioning failed for the template hs-wismar.Message: Connection to device 10.10.20.84 timed out using protocol ssh2; Detail: FAILED : Error in running XDE Procedure, Error Message::Failed to establish ssh2 connection to device- Cause : Connection refused or timed-out.",

Is the device in the inventory and managed?

**Comment by Pratik993** [comment]
Yes @aradford123 , mentioned device is into inventory as i am using reserved sandbox and it is pre-configured.

In DNA center, it shows Reachability Status is unreachable by default. 

I didn't do any changes as just tried to apply VLAN via template

**Comment by aradford123** [comment]
if the device is unreachable, there is not way DNAC can configure it.

Looks like a problem with your sandbox.

**Comment by Pratik993** [comment]
@aradford123 

Thank you for solution sir, but every time, i reserve the sandbox, status is always unreachable by default.

How can i reserve sandbox with reachable inventory?

if any changes need to made in DNAC, Please suggest.
